src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -655,6 +656,10 @@ export class ConnectionDialogWebviewController extends FormWebviewController<
                 cleanedConnection as any,
             );
 
+            if (!cleanedConnection.id) {
```

Comment:
Can you use `ConnectionProfile.addIdIfMissing(profile);` so we can more easily track everywhere that IDs are getting added?  Might help with debugging later.

Reply:
Actually, [this PR](https://github.com/microsoft/vscode-mssql/pull/19137/files) that I currently have out fixes the same issue, but in a more consolidated way (in `connectionStore.saveProfile()`).  I think that approach is better here.